src/o365/graph/commands/teams/teams-add.ts

```diff
@@ -0,0 +1,197 @@
+import auth from '../../GraphAuth';
+import config from '../../../../config';
+import commands from '../../commands';
+import GlobalOptions from '../../../../GlobalOptions';
+import {
+  CommandOption, CommandValidate
+} from '../../../../Command';
+import Utils from '../../../../Utils';
+import GraphCommand from '../../GraphCommand';
+import * as request from 'request-promise-native';
+const vorpal: Vorpal = require('../../../../vorpal-init');
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  groupId?: string;
+  name?: string;
+  description?: string;
+}
+
+class GraphTeamsAddCommand extends GraphCommand {
+  public get name(): string {
+    return `${commands.TEAMS_ADD}`;
+  }
+
+  public get description(): string {
+    return 'Add a new Microsoft Teams team.';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    return telemetryProps;
+  }
+
+  public commandAction(cmd: CommandInstance, args: CommandArgs, cb: () => void): void {
+    auth
+      .ensureAccessToken(auth.service.resource, cmd, this.debug)
+      .then((): request.RequestPromise => {
+        return args.options.groupId ? this.CreateGroupTeamRequest(cmd,args) : 
+                                      this.CreateTeamRequest(cmd,args);
+      })
+      .then((res: any): void => {
+        // get the teams id from the response header.
+        var teamsId = this.GetTeamsIdFromResponse(res, args);
+        cmd.log({Team:teamsId});
+
+        if (this.verbose) {
+          cmd.log(vorpal.chalk.green('DONE'));
+        }
+        cb();
+      }, (err: any): void =>{ 
+        this.handleRejectedODataJsonPromise(err, cmd, cb)
+      });
+  }
+
+  private GetTeamsIdFromResponse(res:any, args: CommandArgs) : string {
```

Comment:
Since we're calling the code from this method from just one place and it's just a couple of lines of code, it would be easier not to define a separate method for it.